metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/NetworkFunctions.java

```diff
@@ -78,13 +78,16 @@ public Object apply(List<Object> list) {
 
     @Override
     public Object apply(List<Object> objects) {
+      if(objects.isEmpty()) {
+        return null;
+      }
       Object dnObj = objects.get(0);
       InternetDomainName idn = toDomainName(dnObj);
       if(idn != null) {
         String dn = dnObj.toString();
         String tld = idn.publicSuffix().toString();
-        String suffix = Iterables.getFirst(Splitter.on(tld).split(dn), null);
-        if(suffix != null)
+        String suffix = dn.substring(0, dn.length() - tld.length());
+        if(suffix != null )
```

Comment:
@james-sirota Can you expand on that?  Here's my understanding, and please correct me if I'm wrong.

www.subdomain.com.com is a valid address.  When we apply the function, we should lose the 'www.subdomain' and leave the domain + '.' + TLD.  In this case the domain is 'com' and the TLD is (coincidentally) 'com', leaving us with 'com.com'.  The unit tests shows this behavior at
` Assert.assertEquals("com.com", removeSubdomains.apply(ImmutableList.of("www.subdomain.com.com")));`